src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -702,7 +702,7 @@ Real SpatialPooler::avgConnectedSpanForColumnND_(UInt column) const {
 void SpatialPooler::adaptSynapses_(const SDR &input,
                                    const SDR &active) {
   for(const auto &column : active.getSparse()) {
-    connections_.adaptSegment(column, input, synPermActiveInc_, synPermInactiveDec_);
+    connections_.adaptSegment(column, input, synPermActiveInc_, synPermInactiveDec_, true);
```

Comment:
SP also does prune synapses (like TM) 
- [ ] default to true/or even rm the parameter, when all use true now